app/controllers/locations_controller.rb

```diff
@@ -0,0 +1,5 @@
+class LocationsController < ApplicationController
+  def index
+    @locations = Locations.near_postcode(params['postcode'])
```

Comment:
oh you mean "no postcode" is the same as "invalid postcode", yeah alright makes sense!